src/datumaro/plugins/configurable_validator.py

```diff
@@ -0,0 +1,732 @@
+        )
+        self.far_from_mean_thr = (
+            far_from_mean_thr if far_from_mean_thr else self.DEFAULT_FAR_FROM_MEAN_THR
+        )
+        self.dominance_thr = (
+            dominance_ratio_thr if dominance_ratio_thr else self.DEFAULT_DOMINANCE_RATIO_THR
+        )
+        self.topk_bins_ratio = topk_bins if topk_bins else self.DEFAULT_TOPK_BINS
+
+    def _init_stats_collector(self, label_categories):
+        for task in self.tasks:
+            if task == TaskType.classification:
+                self.all_stats[task] = ClsStats(
+                    label_categories=label_categories, warnings=self.warnings
+                )
+            elif task == TaskType.detection:
+                self.all_stats[task] = DetStats(
+                    label_categories=label_categories, warnings=self.warnings
+                )
+            elif task == TaskType.segmentation:
+                self.all_stats[task] = SegStats(
+                    label_categories=label_categories, warnings=self.warnings
+                )
+        self.init_flag = True
+
+    def _get_stats_collector(self, ann_type):
+        if TaskType.classification in self.tasks and ann_type == AnnotationType.label:
+            return self.all_stats[TaskType.classification]
+        elif TaskType.detection in self.tasks and ann_type == AnnotationType.bbox:
+            return self.all_stats[TaskType.detection]
+        elif TaskType.segmentation in self.tasks and ann_type in [
+            AnnotationType.mask,
+            AnnotationType.polygon,
+            AnnotationType.ellipse,
+        ]:
+            return self.all_stats[TaskType.segmentation]
+        else:
+            return None
+
+    def compute_statistics(self, dataset):
+        if not self.init_flag:
+            self._label_categories = dataset.categories()[AnnotationType.label]
+            self._init_stats_collector(label_categories=self._label_categories)
+
+        for item in dataset:
+            for stats_collector in self.all_stats.values():
+                if not stats_collector:
+                    continue
+                stats_collector.update_item(item)
+
+            item_key = (item.id, item.subset)
+            for annotation in item.annotations:
+                stats_collector = self._get_stats_collector(ann_type=annotation.type)
+                if not stats_collector:
+                    continue
+                stats_collector.update_ann(item_key, annotation)
+
+        return {task: stats_collector.stats for task, stats_collector in self.all_stats.items()}
+
+    def generate_reports(self, task_stats):
+        reports = {}
+
+        for task, stats in task_stats.items():
+            reports[task] = []
+            if task == TaskType.classification:
+                if {MultiLabelAnnotations} & self.warnings:
+                    reports[task] += self._check_multiple_label(stats)
```

Comment:
Although we produce the reports as `dict`, is it possible to update each part of the dictionary in each collector? I hope that we do not all the things getting together in this function.